scripts/kconfig/menuconfig.py

```diff
@@ -1433,6 +1437,39 @@ def select_prev_match():
             s, s_i, hscroll = _edit_text(c, s, s_i, hscroll,
                                          edit_box.getmaxyx()[1] - 2)
 
+_cached_search_strings = None
+
+def _search_strings():
```

Comment:
You can rewrite this as:

```Python
def _search_strings(cached_search_strings=[]):
   if not cached_search_strings:
       # populate the cache
   return cached_search_strings
```

Reply:
Clever... had never thought of that use for the "interview question" default argument behavior. :P

I prefer to get rid of scopes early, even if it's a bit of extra code, so I think I'll keep the rest as is. I did the other variant in `_style()`.

Reply:
Heh... `_CACHED_NODE_STRINGS = ...`. Blatant bug there. Gets pretty short too, so I'll go for it.

Too much coding over the last few days. :P